### PR TITLE
Add ledger_data RPC tests (RIPD-1260)

### PR DESCRIPTION
**Pull request by mellery451**
Migrate ledger-data-test.js to cpp tests. Provide coverage for paging
limits and input error cases.

This test provides 100% coverage for LedgerData.cpp (RPC handler)

### CODE REVIEW

**Comment by vinniefalco** [comment]
Did you mean ledger_data?

**Comment by mellery451** [comment]
ugh  - yes, correcting

File: src/ripple/rpc/tests/LedgerData.test.cpp [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        Env env(*this, makeConfig(as_admin));
+        Account const gw ("gateway");
+        auto const USD  = gw["USD"];
+        env.fund(XRP(100000), gw);
+
+        const int max_limit = 256; //would be 2048 for binary requests, no need to test that here
+
+        for (auto i = 0; i < max_limit + 10; i++)
+        {
+            Account const bob (std::string("bob") + std::to_string(i));
```
Comment (vinniefalco) [inline src/ripple/rpc/tests/LedgerData.test.cpp]: I proclaim henceforth that all constructor invocations shall use curly braces:
```
Account const bob{std::string("bob") + std::to_string(i)};
```
(except for `std::string`, standard containers, or any constructor that takes a `std::initializer_list`)

**Comment by nbougalis** [comment]
Merged as 8f41817cb9c1a266bd1b502786afef715bb6b74e (typo in commit fixed)

**Pull request closed without merging**